utils.py

```diff
@@ -157,3 +159,22 @@ def balance(it):
 
                 last_label = e[1]
                 yield e
+
+
+def download(url, filename, block='█'):
+    with open(filename, 'wb') as file:
+            response = requests.get(url, stream=True)
+            total = response.headers.get('content-length')
+
+            if total is None:
+                file.write(response.content)
+            else:
+                downloaded = 0
+                total = int(total)
+                for data in response.iter_content(chunk_size=int(total / 1000)):
```

Comment:
Maybe the chunk size is a little too big? If total is 1 GB, the chunk size is 100 MB.

Reply:
I'm dividing the total size by 1000, so for a 1 GB file, it's around 1 MB, which I think is pretty optimum. I went ahead and made it `max(int(total / 1000), 1024 * 1024)`, so that for small files the chunk size doesn't get too small.

Reply:
Sounds good. We might also want to introduce a maximum, but it's fine for now.